bots/elasticsearch/bot.py

```diff
@@ -79,67 +79,67 @@ class ElasticSearchBot(Bot):
-                'description': {'type': 'string'},
-                'funders': {'type': 'string', 'fields': EXACT_FIELD},
-                'hosts': {'type': 'string', 'fields': EXACT_FIELD},
-                'id': {'type': 'string', 'index': 'not_analyzed', 'include_in_all': False},
-                'identifiers': {'type': 'string', 'fields': EXACT_FIELD},
-                'justification': {'type': 'string', 'include_in_all': False},
-                'language': {'type': 'string', 'index': 'not_analyzed', 'include_in_all': False},
-                'publishers': {'type': 'string', 'fields': EXACT_FIELD},
-                'registration_type': {'type': 'string', 'index': 'not_analyzed', 'include_in_all': False},
+                'description': {'type': 'text'},
+                'funders': {'type': 'text', 'fields': EXACT_FIELD},
+                'hosts': {'type': 'text', 'fields': EXACT_FIELD},
+                'id': {'type': 'text', 'index': 'not_analyzed', 'include_in_all': False},
+                'identifiers': {'type': 'text', 'fields': EXACT_FIELD},
+                'justification': {'type': 'text', 'include_in_all': False},
+                'language': {'type': 'keyword', 'include_in_all': False},
+                'publishers': {'type': 'text', 'fields': EXACT_FIELD},
+                'registration_type': {'type': 'keyword', 'include_in_all': False},
                 'retracted': {'type': 'boolean', 'include_in_all': False},
-                'sources': {'type': 'string', 'index': 'not_analyzed', 'include_in_all': False},
-                'subjects': {'type': 'string', 'index': 'not_analyzed', 'include_in_all': False},
-                'tags': {'type': 'string', 'fields': EXACT_FIELD},
-                'title': {'type': 'string', 'fields': EXACT_FIELD},
-                'type': {'type': 'string', 'index': 'not_analyzed', 'include_in_all': False},
-                'types': {'type': 'string', 'index': 'not_analyzed', 'include_in_all': False},
+                'sources': {'type': 'keyword', 'include_in_all': False},
+                'subjects': {'type': 'keyword', 'include_in_all': False},
+                'tags': {'type': 'text', 'fields': EXACT_FIELD},
+                'title': {'type': 'text', 'fields': EXACT_FIELD},
+                'type': {'type': 'keyword', 'include_in_all': False},
+                'types': {'type': 'keyword', 'include_in_all': False},
                 'withdrawn': {'type': 'boolean', 'include_in_all': False},
                 'lists': {'type': 'object', 'dynamic': True, 'include_in_all': False},
             },
             'dynamic_templates': [
-                {'exact_field_on_lists_strings': {'path_match': 'lists.*', 'match_mapping_type': 'string', 'mapping': {'type': 'string', 'fields': EXACT_FIELD}}},
+                {'exact_field_on_lists_strings': {'path_match': 'lists.*', 'match_mapping_type': 'text', 'mapping': {'type': 'text', 'fields': EXACT_FIELD}}},
```

Comment:
This [should still be `'string'`](https://www.elastic.co/guide/en/elasticsearch/reference/5.4/dynamic-templates.html#match-mapping-type)